src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -186,7 +186,7 @@ private Guid LaunchAgentProcess(TestPackage package)
             if (debugAgent)
                 agentArgs += " --debug-agent";
             if (workDirectory != string.Empty)
-                agentArgs += " --work=" + workDirectory;
+                agentArgs += " --work \"" + workDirectory + "\"";
```

Comment:
This isn't intended to be run by users. It's only executed from Test Agency.

Reply:
Ah ok. Thanks for explaining!

Reply:
In the current version, even if user tries to pass a quoted working directory to the nunit3-console.exe, the console tool itself gets the correct path (value of the workDirectory variable), but when it will try to run the agent using the code above - the path will become invalid. My fix solves that issue.

Reply:
That's because quotes are stripped from the command-line before being passed to the program.